src/main/java/com/flightstats/hub/channel/ChannelStatusResource.java

```diff
@@ -31,10 +31,11 @@
     public Response getLatest(@PathParam("channel") String channel,
                               @QueryParam("stable") @DefaultValue("true") boolean stable,
                               @QueryParam("trace") @DefaultValue("false") boolean trace) {
-        ChannelConfig channelConfig = channelService.getCachedChannelConfig(channel);
-        if (null == channelConfig) {
+        Optional<ChannelConfig> optionalChannelConfig = channelService.getCachedChannelConfig(channel);
```

Comment:
I would probably extract the stuff from line 38 onwards into another method that takes a `ChannelConfig` and whatever else it needs, which'd make this a 3 or 4 line method, but... the rest of this method is not straightforward and probably needs more love than just that. So forget I said anything.